sdk/eventhubs/azure-messaging-eventhubs/src/samples/java/com/azure/messaging/eventhubs/EventHubProducerAsyncClientJavaDocCodeSamples.java

```diff
@@ -37,84 +35,108 @@ public void instantiate() {
     }
 
     /**
-     * Code snippet demonstrating how to create an {@link EventHubProducerAsyncClient} that routes events to a single
-     * partition.
-     *
+     * Code snippet demonstrating how to send a batch that automatically routes events to any partition.
      */
-    public void instantiatePartitionProducer() {
-        // BEGIN: com.azure.messaging.eventhubs.eventhubasyncproducerclient.instantiation#partitionId
-        EventData eventData = new EventData("data-to-partition-foo");
-        SendOptions options = new SendOptions()
-            .setPartitionId("foo");
+    public void batchAutomaticRouting() {
+        // BEGIN: com.azure.messaging.eventhubs.eventhubasyncproducerclient.createBatch
+        // The required parameter is a way to authenticate with Event Hubs using credentials.
+        // The connectionString provides a way to authenticate with Event Hub.
+        EventHubProducerAsyncClient producer = new EventHubClientBuilder()
+            .connectionString(
+                "Endpoint={fully-qualified-namespace};SharedAccessKeyName={policy-name};SharedAccessKey={key}",
+                "event-hub-name")
+            .buildAsyncProducerClient();
+
+        // Creating a batch without options set, will allow for automatic routing of events to any partition.
+        producer.createBatch().flatMap(batch -> {
+            batch.tryAdd(new EventData("test-event-1"));
+            batch.tryAdd(new EventData("test-event-2"));
+            return producer.send(batch);
+        }).block();
```

Comment:
Should our samples have `block()` or `subscribe()` for async clients?

Reply:
Hmm... The bad thing is if I use .subscribe, since it's a non-blocking call, we'd have to add some sort of "Thread.sleep()" is that what we want instead?

Reply:
Yeah, it's tricky. Users usually copy samples and they'll block which defeats the purpose of async. If you add sleep, it's not ideal but at least users will realize they want to do something else here instead of sleep. Hopefully, adding a comment before sleep will help them understand what is expected.